railties/lib/rails/generators/generated_attribute.rb

```diff
@@ -118,12 +120,12 @@ def parse_attr_options(options_str)
           end
 
           def parse_attr_option(option)
-            if option.match(/(.+)=\{(.+)\}/)
+            if option.match(/(.+):\{(.+)\}/)
               key, nested_opt = $1, $2
 
               [key, Hash[*parse_attr_option(nested_opt)]]
```

Comment:
I think we could use `to_h` here as an alternative.

```suggestion
              [key, parse_attr_option(nested_opt).to_h]
```

Reply:
`parse_attr_option` will return a tuple. `to_h` needs an array of tuples.

Reply:
I had to look this up in the [docs](https://docs.ruby-lang.org/en/master/Hash.html#method-c-5B-5D), so transcribing here for anyone else:

> When the argument count is an even number; returns a new Hash object wherein each successive pair of arguments has become a key-value entry:
> 
> ```ruby
> Hash[:foo, 0, :bar, 1] # => {:foo=>0, :bar=>1}
> ```

TIL